File: 2215.java
//* @author:
import java.io.*;
import java.util.*;
public class Main {
   static int SZ=1005;
   public static void main(String[] args) throws Exception {
        int r,c,q;
        int sum;
        int i,j,x1,y1,x2,y2;
    int[][] area=new int[SZ][SZ];
    for(i=0;i< SZ;++i) area[i][0]=0;
    Scanner cin=new Scanner(System.in);
        int T=cin.nextInt();
    while(--T>=0) {
         r=cin.nextInt();
      c=cin.nextInt();
      for(i=0;i< r;++i)
           for(j=1;j<=c;++j){
        area[i][j]=cin.nextInt();
        area[i][j]+=area[i][j-1];
       }
       q=cin.nextInt();
       while(--q>=0)
           {
        x1=cin.nextInt();
        y1=cin.nextInt();
        x2=cin.nextInt();
        y2=cin.nextInt();
        sum=0;
        for(i=x1-1;i< x2;++i)
            sum+=area[i][y2]-area[i][y1-1];
              System.out.println("Absolutni hodnota pohodlnosti je "+sum+" bodu.");
            }
        System.out.println("");
        }
 }
}

File: 2215.cc
nclude<iostream>
#include<cstdio>
using namespace std;

int maze[1010][1010];

int main()
{
    int T;
    scanf("%d",&T);
    while(T--)
    {
        int i,j,m,n;
        scanf("%d%d",&m,&n);
        for(i=0;i<m;i++)
            for(j=0;j<n;j++)
                scanf("%d",&maze[i][j]);
        int t;
        scanf("%d",&t);
        while(t--)
        {
            int ans=0;
            int r1,r2,s1,s2;
            scanf("%d%d%d%d",&r1,&r2,&s1,&s2);
            for(i=r1-1;i<s1;i++)
                for(j=r2-1;j<s2;j++)
                    ans+=maze[i][j];
            printf("Absolutni hodnota pohodlnosti je %d bodu.\n",ans);
        }
        if(T)
            cout<<endl;
    }
    return 0;
}



